### PR TITLE
refacto randiter (seed first)

### PR DESCRIPTION
**Pull request by jgdumas**
1/ seed before size
2/ size is sampling size (same type as cardinality) and should be used

### CODE REVIEW

File: tests/test-matpoly-mult.C [hunk 1]
```diff
@@ -175,7 +175,7 @@ bool debug_midpgen_dlp(const Field& fld,  RandIter& Gen) {
 template<typename Field>
 bool launchTest(const Field& F, size_t n, long b, long d, long seed){
 	bool ok=true;
-	typename Field::RandIter G(F,b,seed);
+	typename Field::RandIter G(F,seed,b);
```
Comment (ClementPernet) [inline tests/test-matpoly-mult.C]: b is a bitsize: should use a samplesize(1); samplesize<<b;

File: tests/test-solve-full.C [hunk 2]
```diff
@@ -59,7 +59,7 @@ namespace {
     void generateMatrix(Domain& D, Matrix& A, int bitSize, int seed)
     {
         // @note RandomDenseMatrix can also fill a SparseMatrix, it has just a sparsity of 0.
-        typename Domain::RandIter randIter(D, bitSize, seed);
+        typename Domain::RandIter randIter(D, seed, bitSize);
```
Comment (ClementPernet) [inline tests/test-solve-full.C]: same her

File: tests/test-solve-full.C [hunk 3]
```diff
@@ -75,7 +75,7 @@ namespace {
     template <class Matrix, class Vector, class Domain>
     void generateVector(Domain& D, const Matrix& A, Vector& b, int vectorBitSize, int seed)
     {
-        typename Domain::RandIter vectorRandIter(D, vectorBitSize, seed);
+        typename Domain::RandIter vectorRandIter(D, seed, vectorBitSize);
```
Comment (ClementPernet) [inline tests/test-solve-full.C]: same here

File: tests/test-matpoly-mult.C (hunk 1 above)
Comment (jgdumas) [reply tests/test-matpoly-mult.C]: done

File: tests/test-solve-full.C (hunk 2 above)
Comment (jgdumas) [reply tests/test-solve-full.C]: done

File: tests/test-solve-full.C (hunk 3 above)
Comment (jgdumas) [reply tests/test-solve-full.C]: done

**Comment by ClementPernet** [comment]
LGTM

**Pull request merged**